src/lib/dom-api-mutation-content.html

```diff
@@ -0,0 +1,111 @@
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="settings.html">
+<script>
+(function() {
+  'use strict';
+
+    var DomApi = Polymer.DomApi.ctor;
+    var Settings = Polymer.Settings;
+
+    DomApi.MutationContent = function(domApi) {
+      DomApi.Mutation.call(this, domApi);
+    };
+
+    DomApi.MutationContent.prototype = Object.create(DomApi.Mutation.prototype);
+
+    Polymer.Base.extend(DomApi.MutationContent.prototype, {
+
+      addListener: function(callback, options) {
+        var h = DomApi.Mutation.prototype.addListener.call(this, callback, 
+          options);
+        this._scheduleNotify();
+        return h;
+      },
+
+      _ensureSetup: function(options) {},
+
+      notify: function() {
+        if (this._hasListeners()) {
+          this._scheduleNotify();
+        }
+      },
+
+      _notify: function() {
+        var info = this._suspendChangeInfo ? {} : this._calcChanges();
+        if (info) {
+          info.target = this.node;
+          this._callListeners(info);
+        }
+      },
+
+      _calcChanges: function() {
+        var changes = {
+          addedNodes: [],
+          removedNodes: []
+        };
+        var o$ = this.node.__distributedNodes = this.node.__distributedNodes ||
+          [];
+        var n$ = this.domApi.getDistributedNodes();
+        this.node.__distributedNodes = n$;
+        var splices = Polymer.ArraySplice.calculateSplices(n$, o$);
+        // process removals
+        for (var i=0, s; (i<splices.length) && (s=splices[i]); i++) {
+          for (var j=0, n; (j < s.removed.length) && (n=s.removed[j]); j++) {
+            changes.removedNodes.push(n);
+          }
+        }
+        // process adds
+        for (var i=0, s; (i<splices.length) && (s=splices[i]); i++) {
+          for (var j=s.index; j < s.index + s.addedCount; j++) {
+            changes.addedNodes.push(n$[j]);
+          }
+        }
+        if (changes.addedNodes.length || changes.removedNodes.length) {
+          return changes;
+        }
+      }
+
+    });
+
+    if (Settings.useShadow) {
+      
+      Polymer.Base.extend(DomApi.MutationContent.prototype, {
+        
+        _ensureSetup: function(options) {
+          this._trackAttributes = this._trackAttributes ||
+            options.attributes;
+          if (!this._observer) {
+            var root = this.domApi.getOwnerRoot();
+            var host = root && root.host;
+            if (host) {
+              this._observer = Polymer.dom(host).observeNodes(
+                this.notify.bind(this), {attributes: this._trackAttributes});
+            }

```

Comment:
Will not recurse back up with attributes:true if was called previously with attributes:false, since ensureSetup is 1-shot